Source/Core/DolphinQt/Settings.cpp

```diff
@@ -126,19 +129,45 @@ QString Settings::GetCurrentUserStyle() const
 }
 
 // Calling this before the main window has been created breaks the style of some widgets.
-void Settings::SetCurrentUserStyle(const QString& stylesheet_name)
+void Settings::SetCurrentUserStyle(QString stylesheet_name)
 {
+#ifdef _WIN32
+  // On the first boot of Dolphin, automatically select the dark theme
+  // if the Windows system has it enabled
+  if (stylesheet_name.isEmpty())
+  {
+    using namespace winrt::Windows::UI::ViewManagement;
+    const UISettings settings;
+    const auto& color = settings.GetColorValue(UIColorType::Foreground);
+
+    const bool is_system_dark = (((5 * color.G) + (2 * color.R) + color.B) > (8 * 128));
+    if (is_system_dark)
+    {
+      stylesheet_name = QStringLiteral("Dark");
+    }
+  }
+#endif
+
   QString stylesheet_contents;
 
   // If we haven't found one, we continue with an empty (default) style
-  if (!stylesheet_name.isEmpty() && AreUserStylesEnabled())
+  if (!stylesheet_name.isEmpty())
   {
-    // Load custom user stylesheet
-    QDir directory = QDir(QString::fromStdString(File::GetUserPath(D_STYLES_IDX)));
-    QFile stylesheet(directory.filePath(stylesheet_name));
+    const auto& stylesheet_path = File::GetStylesDir(stylesheet_name.toStdString());
+    if (stylesheet_path)
+    {
+      // Load stylesheet
+      QDir directory = QDir(QString::fromStdString(stylesheet_path.value()));
+      QFile stylesheet(directory.filePath(QStringLiteral("style.qss")));
+      QString resource_path = directory.filePath(QStringLiteral("style.rcc"));
+
+      if (stylesheet.open(QFile::ReadOnly))
+        stylesheet_contents = QString::fromUtf8(stylesheet.readAll().data());
 
-    if (stylesheet.open(QFile::ReadOnly))
-      stylesheet_contents = QString::fromUtf8(stylesheet.readAll().data());
+      // Register QResource file if it exists
+      if (QFile(resource_path).exists())
+        QResource::registerResource(resource_path);
```

Comment:
This seems really janky. If you want to support loading .rcc files, you should load the entire stylesheet from it, not just the images. You should also unregister the resource when the style is switched again, as you can't guarantee multiple styles won't share the same image filenames. The `mapRoot` parameter to `registerResource()` should also probably be set to something that is guaranteed non-conflicting with other resources.

Reply:
In fact, this seems complex enough to handle correctly that I'd prefer the stylesheet loading logic changes in a separate PR.

Reply:
I agree. The changes have been removed from this PR.